Assets/Scripts/LogicServices/AlgorithmService.cs

```diff
@@ -11,18 +11,27 @@ public class AlgorithmService
     {
         private readonly SelfBuild _selfBuild;
         private readonly Randomness _randomness;
+        private readonly BuildEffect _buildEffect;
 
         public AlgorithmService()
         {
             Random random = new Random();
             _selfBuild = new SelfBuild(random);
             _randomness = new Randomness(random);
+            _buildEffect = new BuildEffect(random);
         }
 
         public bool Decide(Trade trade,Tribe originator, Tribe targetCpu)
         {
-            if (_randomness.Calculate()) return !_selfBuild.Calculate(trade,targetCpu);
-            return _selfBuild.Calculate(trade,targetCpu);
+            //Original Decisions
+            bool selfBuildDecision = _selfBuild.Calculate(trade, targetCpu);
+            bool buildEffectDecision = _buildEffect.Calculate(trade, targetCpu, originator);
+            
+            //Randomise Decisions
+            selfBuildDecision = _randomness.Calculate(0.2f) ? selfBuildDecision : !selfBuildDecision;
+            buildEffectDecision = _randomness.Calculate(0.1f) ? buildEffectDecision : !buildEffectDecision;

```

Comment:
I'd probably prefer to have these randomness variables somewhere up top for easy access and tweaking, perhaps possibly even in editor.

Reply:
Yea, I agree, I would also prefer to use a variable instead of these magic numbers.